test/test_prototype_transforms_functional.py

```diff
@@ -1103,22 +1112,67 @@ def test_correctness_pad_segmentation_mask_on_fixed_input(device):
     torch.testing.assert_close(out_mask, expected_mask)
 
 
+def _parse_padding(padding):
+    if isinstance(padding, int):
+        return [padding] * 4
+    if isinstance(padding, list):
+        if len(padding) == 1:
+            return padding * 4
+        if len(padding) == 2:
+            return padding * 2  # [left, up, right, down]
+
+    return padding
+
+
+@pytest.mark.parametrize("device", cpu_and_gpu())
+@pytest.mark.parametrize("padding", [[1], [1, 1], [1, 1, 2, 2]])
+def test_correctness_pad_bounding_box(device, padding):
+    def _compute_expected_bbox(bbox, padding_):
+        pad_left, pad_up, _, _ = _parse_padding(padding_)
+
+        bbox_format = bbox.format
+        bbox_dtype = bbox.dtype
+        bbox = convert_bounding_box_format(bbox, old_format=bbox_format, new_format=features.BoundingBoxFormat.XYXY)
+
+        bbox[0::2] += pad_left
+        bbox[1::2] += pad_up
+
+        bbox = convert_bounding_box_format(
+            bbox, old_format=features.BoundingBoxFormat.XYXY, new_format=bbox_format, copy=False
+        )
+        if bbox.dtype != bbox_dtype:
+            # Temporary cast to original dtype
+            # e.g. float32 -> int
+            bbox = bbox.to(bbox_dtype)
+        return bbox
```

Comment:
1. There is no need for this `if`: `.to` is a no-op if no conversion needs to be performed.
2. You don't need to use the dtype explicitly. You can pass a tensor to `.to` in which case it will convert device and dtype to the input.

```suggestion
        return convert_bounding_box_format(
            bbox, old_format=features.BoundingBoxFormat.XYXY, new_format=bbox_.format, copy=False
        ).to(bbox_)
```

Reply:
@pmeier actually it does not work with `.to` between tensors and BoundingBox:
```python
>>> t = torch.tensor([[1.0, 2.0, 3.0, 4.0]])
>>> bbox = features.BoundingBox([[0, 1, 2, 3]], format=features.BoundingBoxFormat.XYXY, image_size=(32, 32))
>>> t.to(bbox)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/vision/torchvision/prototype/features/_feature.py", line 83, in __torch_function__
    return cls.new_like(args[0], output, dtype=output.dtype, device=output.device)
  File "/vision/torchvision/prototype/features/_bounding_box.py", line 54, in new_like
    format=format if format is not None else other.format,
AttributeError: 'Tensor' object has no attribute 'format'
```

Reply:
I'll added #6094 to track this.